### ISSUE TITLE
headerBackTitle not working

### ISSUE THREAD

**Issue by PierreMartin** [react-navigation/react-navigation#11375]
### Current behavior

In a screen, I want to have a custom text on the back button, I use headerBackTitle.

```js
  React.useLayoutEffect(() => {
        navigation.setOptions({
          title: "Title screen",
          headerBackTitle: "Go back..."
    });
}, []);

```

![Simulator Screenshot - iPhone SE (3rd generation) - 2023-05-17 at 19 30 36](https://github.com/react-navigation/react-navigation/assets/12272752/23347c14-6919-414b-a2e6-9f5e47b4089f)

### Expected behavior

I should have the content of  headerBackTitle: "Go back..." in the back button

 

### Reproduction

https://github.com/PierreMartin/AwesomeProject

### Platform

- [ ] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [X] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [X] @react-navigation/stack
- [X] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [X] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | 6.1.6
| @react-navigation/bottom-tabs          | 6.5.7
| @react-navigation/material-top-tabs    | 6.6.2
| @react-navigation/stack                | 6.3.16
| @react-navigation/native-stack         | 6.9.12
| react-native-safe-area-context         | 4.5.1
| react-native-screens                   | 3.20.0
| react-native-gesture-handler           | 2.9.0
| react-native-reanimated                | 2.12.0
| react-native-tab-view                  | 3.5.1
| react-native-pager-view                | 6.2.0
| react-native                           | 0.71.7
| node                                   | 18.15.0
| npm                            | 9.5.0

**Comment by Jmzp** [comment]
Same here, I have the same problem but in Android

**Comment by BeeMargarida** [comment]
> Same here, I have the same problem but in Android

This is expected not to work in Android, `headerBackTitle` is only supported for iOS - https://reactnavigation.org/docs/native-stack-navigator#headerbacktitle

**Comment by bbialas** [comment]
I have the same issue, app ignores some properties, for example `headerBackTitle` and `headerBackTitleVisible`.
I'm using expo-router 1.5.3 and react-navigation native-stack in version 6.9.12

**Comment by Blackfaded** [comment]
Having the same issue on ios
```ts
export const LoginStackScreen = () => {
  const { colors } = useTheme();
  return (
    <LoginStack.Navigator>
      <LoginStack.Group
        screenOptions={{
          ...getSharedHeaderStyle(colors),
        }}
      >
        <LoginStack.Screen
          options={{ title: 'Anmelden', headerBackTitle: 'Zurück' }}
          name={Screens.LOGIN}
          component={LoginScreen}
        />
        <LoginStack.Screen
          options={{ title: 'Passwort wiederherstellen', headerBackTitle: 'Zurück' }}
          name={Screens.FORGET_PASSWORD}
          component={ForgetPasswordScreen}
        />
        <LoginStack.Screen
          options={{ title: 'Registrieren', headerBackTitle: 'Zurück' }}
          name={Screens.REGISTER}
          component={RegisterScreen}
        />
      </LoginStack.Group>
    </LoginStack.Navigator>
  );
};
```
Results in 
<img width="435" alt="Bildschirmfoto 2023-05-31 um 22 34 22" src="https://github.com/react-navigation/react-navigation/assets/19570581/d20801c5-4b7a-4ba6-a1e6-7a9628a4e565">

**Comment by holakmateusz** [comment]
I have the same problem, it looks like headerBackTitle does not reflect anything on iOS.

**Comment by ConnorMarble** [comment]
Same problem on iOS for me.

**Comment by ibrahimshittu** [comment]
same issue son IOS

**Comment by itajenglish** [comment]
+1

**Comment by paule89123** [comment]
+1

**Comment by rKatex** [comment]
Same problem here. It also seems that `headerBackTitle` isn't working eather on iOS.

```json
    "@react-navigation/native": "6.1.6",
    "@react-navigation/native-stack": "6.9.12",
```

**Comment by ronweli** [comment]
same problem. headerBackTitleVisible: false doesn't hide the 'Back' text in iOS header.

 "@react-navigation/bottom-tabs": "^6.5.7",
    "@react-navigation/native": "^6.1.6",
    "@react-navigation/native-stack": "^6.9.12",
    "@react-navigation/stack": "^6.3.16",
    "react": "18.2.0",
    "react-native": "0.71.8",

**Comment by dreson4** [comment]
I think it's a bug with header back title on iOS.
The way it currently work is if you have a stack, the first screen must have header, if headerShown is false on the first screen of the stack then the next screen headerBackTitle settings won't work. 

The only way to have headerBackTitleVisible and headerBackTitle to work is to have the first screen on your stack have headerShown true which sucks because on tabs for example I would like to hide the header usually. 

An example of code that won't work, on Other Screens, if you have headerShown to true there will always be Back title, there's no way to override that, headerBackTitleVisible:false, headerBackTitle:"something" it won't work. 

```
 <ProfileStack.Navigator>
            <ProfileStack.Group screenOptions={{headerBackTitleVisible: true, headerBackTitle: 'test'}}>
                <ProfileStack.Screen
                    name={'Profile'}
                    component={Profile}
                    options={({route}) => ({headerShown:false})}}
                />
                /* other screens */
            </ProfileStack.Group>
        </ProfileStack.Navigator>
```

**Comment by kacperkapusciak** [comment]
Fixed in `react-native-screens@3.21.1` and `@react-navigation/native-stack@6.9.13`. 

Cheers ✌️

**Comment by jxxe** [comment]
@dreson4 Did you ever find a solution to the back button text not working if the previous header is hidden?

**Comment by timothyerwin** [comment]
> headerBackTitleVisible

you just saved me there. I wasn't trying to figure out how to turn off the back button text and this did it. awesome!

**Comment by Sarinduu** [comment]
It works for me on iOS when I use `headerTitle` instead of `headerBackTitle`

```
 <Stack.Screen
          name="user profile"
          component={Profile}
          options={{ title : "profile" , headerTitle : "back"}}
        />
```

**Comment by Marc-9** [comment]
> Fixed in `react-native-screens@3.21.1` and `@react-navigation/native-stack@6.9.13`.
> 
> Cheers ✌️

Unfortunately this fix only works in the case where you want to replace the HeaderBackTitle with something else

If I want to set HeaderBackTitle to an empty string it reverts back to using the screen name

**Comment by Elue-dev** [comment]
in case anyone is experiencing this, use `headerTintColor` instead
eg: `headerTintColor: "#fff",`
